doc/design_AsyncNDRegisterAcessor_and_NumericAddressedBackend.dox

```diff
@@ -0,0 +1,60 @@
+</pre>
+This function has to be implemented by the manager. From the ID it knows which subscriber is to be removed. For this mechanism to work, the AsyncAccessorManager must store boost::weak_pointers and lock them when the instance is being accessed. Otherwise the one instance held within the manager prevents the destructor from being called.
+
+\section  design_async_NumericAddressedBackend Asynchronous accessors in the NumericAddressedBackend
+
+As the NumericAddressedBackend base class already provides the creation of the synchronous NumericAddressedBackendRegisterAccessors, it also provides the subscription mechanism for the asynchronous accessors. The NumericAddressedBackend is based on a map file which allows to describe registers with interrupts. In the interface to the hardware, all interrupts are void. The map file then descibes which of the synchronous registers are read out and pushed when the according interrupt is triggered. This mechanism is implemented in the NumericAddressedBackend base class and the protocol implementing backends only have to notify the NumericAddressedInterruptDispatcher that the interrupt has happened.
+
+\subsection  async_NumericAddressedBackend_interface Interface for backend implementations
+
+The NumericAddressedInterruptDispatcher serves two purposes:
+\li It serves as a subscription manager for the AsyncNDRegisterAccessor
+\li It provides functions which act on all subsribers.
+
+When the map file is read in, a NumericAddressedInterruptDispatcher is created for each interrupt.
+
+The only function the backend implementor really has to deal with is NumericAddressedInterruptDispatcher::trigger(). All other functionaltiy for subscription, activation/deactivation and exception handling is already handled in the NumericAddressedBackend.
+
+There is one NumericAddressedInterruptDispatcher per interrupt controller number and interrupt number, stored in the variable NumericAddressedBackend::_interruptDispatchers. The interrupt handling thread in the backend implementation will just call
+
+<pre>
+  _interruptDispatcher.at({interruptControllerNumber, interruptNumber})->trigger();
+</pre>
+
+Notice that you should access the map with at() and not through the [ ] operator, so you don't accidentally add empty dispatchers to the map.
+The interrupt handler thread should have a "lazy start", which means that it should only be started once accessors to the particular interrupt have been requested. To achieve this, the backend should override NumericAddressedBackend::startInterruptHandlingThread(unsigned int interruptControllerNumber, unsigned int interruptNumber). This function is called each time an asynchronous accessor which is associated to that interrupt is created. The implementing code has to make sure that the thead is only started once. The thread in not terminated automatically. The backend has to take care to do this when the backend is closed and/or destructed.
```

Comment:
thead -> thread

Reply:
"The thread in not" -> "The thread is not"